Source/EkoActionSheetViewController.m

```diff
@@ -148,62 +152,118 @@ - (void)configureView
-    
     self.titleLabel.text = self.actionSheetTitle ? self.actionSheetTitle : NSLocalizedString(@"", nil);
-
-    NSString *cancelTitle = self.cancelButtonTitle ? self.cancelButtonTitle : NSLocalizedString(@"Cancel", nil);
-    [self.cancelButton setTitle:cancelTitle forState:UIControlStateNormal];
-    [self.cancelButton addTarget:self action:@selector(cancelButtonDidTouched:) forControlEvents:UIControlEventTouchUpInside];
-    self.cancelButton.hidden = self.cancelButtonHidden;
     
-    NSString *applyTitle = self.applyButtonTitle ? self.applyButtonTitle : NSLocalizedString(@"Apply", nil);
-    [self.applyButton setTitle:applyTitle forState:UIControlStateNormal];
-    [self.applyButton addTarget:self action:@selector(applyButtonDidTouched:) forControlEvents:UIControlEventTouchUpInside];
-    self.applyButton.hidden = self.applyButtonHidden;
+    if (self.cancelItem) {
+        [self.cancelButton setTitle:self.cancelItem.title forState:UIControlStateNormal];
+        [self.cancelButton addTarget:self action:@selector(cancelItemActionHandler) forControlEvents:UIControlEventTouchUpInside];
+    } else {
+        self.cancelButton.hidden = YES;
+    }
+    
+    if (self.applyItem) {
+        [self.applyButton setTitle:self.applyItem.title forState:UIControlStateNormal];
+        [self.applyButton addTarget:self action:@selector(applyItemActionHandler) forControlEvents:UIControlEventTouchUpInside];
+    } else {
+        self.applyButton.hidden = YES;
+    }
+    
+    NSString *cellNibName = NSStringFromClass([EkoActionSheetCell class]);
+    UINib *cellNib = [UINib nibWithNibName:cellNibName bundle:[NSBundle bundleForClass:[self class]]];
+    [self.tableView registerNib:cellNib forCellReuseIdentifier:cellNibName];
+    
+    for (EkoActionSheetItem *item in self.items) {
+        if (item.cellNibName && item.cellIdentifier) {
+            UINib *customCellNib = [UINib nibWithNibName:item.cellIdentifier bundle:nil];
+            [self.tableView registerNib:customCellNib forCellReuseIdentifier:item.cellIdentifier];
+        }
+    }
 }
 
 - (CGFloat)visibleViewHeight
 {
     CGFloat headerHeight = kEkoActionSheetHeaderViewHeight;
-    CGFloat cellHeight = kEkoActionSheetCellHeight;
-    NSUInteger cellCount = self.dataSource.numberOfActions;
-    CGFloat paddingBottom = 0;
-    if ([self.dataSource respondsToSelector:@selector(actionSheetPaddingBottom)]) {
-        paddingBottom = [self.dataSource actionSheetPaddingBottom];
+    CGFloat rowHeight = kEkoActionSheetCellHeight;
+    
+    CGFloat tableHeight = 0;
+    for (EkoActionSheetItem *item in self.items) {
+        if (item.heightBlock) {
+            tableHeight += item.heightBlock();
+        } else {
+            tableHeight += rowHeight;
+        }
     }
     
-    return ceilf(headerHeight + cellHeight * cellCount + paddingBottom);
+    return ceilf(headerHeight + tableHeight);
```

Comment:
correct. recommended to use round values for the view frames, if possible, to avoid side effect issues and for better performance.